documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,295 @@
+
+This documents defines that Camara OpenId Providers MUST ignore the parameter acr_values. 
+
+This document defines that Camara Clients SHOULD not use the acr_values parameter. 
+
+> To foster interoperability a future version of this document might define values for the acr_values parameter acceptable in Camara.
+
+## Access Token Request
+
+This CAMARA document makes the scope parameter REQUIRED for the OAuth2 Client Credentials Grant.
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## The Scope Parameter
+
+
+Scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services. The scope values must be documented in the API OAS files by all Camara API subprojects. This document does not change OIDC definitions of scope values.
+
+
+---
+**NOTE**
+
+Scope values are an integral part of any OAuth2 and OIDC implementation. The RS enforces API access based on scope (if the Camara API subproject defines scopes).
+Therefore scopes should be available to API implementations.
+
+---
+
+
+## Missing "openid" scope
+
+[OIDC Core Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) states the following about the value of scope.
+> REQUIRED. OpenID Connect requests MUST contain the openid scope value. If the openid scope value is not present, the behavior is entirely unspecified. Other scope values MAY be present. ...
+
+This document defines that the authentication server SHOULD not return an id token if `openid` is missing in the scope parameter.
+
+This document defines the following error handling for a missing "openid" value in scope. 
+Please refer to [Authentication Error Response](https://openid.net/specs/openid-connect-core-1_0.html#AuthError).
+
+If "openid" is missing in the scope value but a claim that is [standardized in OIDC](https://openid.net/specs/openid-connect-core-1_0.html#Claims) is requested, then the Authorization Server returns an HTTP response code of 400 (Bad Request) and an error invalid_request.
+
+Clients SHOULD follow the OIDC standard and SHOULD include `openid` in the list of requested scopes.
+Without id token there is no `sub` field and the privacy features of OIDC are severely crippled. 
+Globally unique identifiers, like the MSISDN, should be avoided for privacy reasons.
+
+## Purpose
+
+In Camara `purpose` is used to convey to the user for which purpose an API is used. E.g.: Does the client request the user's location because it wants to show relevant tourist information or is the user's location informat protecting the user when they withdraw money from an ATM?
+Therefore `purpose`is only used in Camara when user consent is required (CIBA or OIDC authorization code flow) and a potential `prompt` parameter's value is NOT `none`.
+
```

Comment:
please replace line #215 by the text below:

Therefore `purpose` is only used in Camara when privacy sensitive data is subject to the API invocation and  explicit check for user consent must happen. This consent check is applicable only for CIBA or OIDC authorization code flow, where the subject of the API call is known at access token request. 

Reason is that we need to send the purpose on all CIBA and auth code flows, where we handle potentially privacy sensitive data. this does not mean that the handling is always subject to user consent. Please replace line 215 with suggestion above.
Also, please remove the statement about the prompt parameter (because it is wrong for legal base = legitimate interest). There might be a consent flow needed or not. This is a topic on its own and needs time to discuss.

Reply:
Sorry for the late finding, but we need the purpose sent on all of the auth code flow and CIBA requests for Camara APIs.

Reply:
From TEF point of view, we can't accept that sentence as it stands either. This topic is discussed further in #131 and TEF's position on when "purpose" must be declared is given [here](https://github.com/camaraproject/IdentityAndConsentManagement/issues/131#issuecomment-1968441327).

>The CAMARA community agreement on 2-legged vs. 3-legged access can be summarised (as documented) as "Every time personal user data is processed by an API and the user can exercise his rights either via opt-in and/or opt-out, 3-legged access tokens must be used."
>
>However, this is an independent debate on whether the purpose for which personal data is consumed should be declared when using client credentials (2-legged access).
>
>From Telefónica's point of view, when personal data is processed, the purpose must always be declared
(this is what the GDPR requires), regardless of the legal basis and whether it requires opt-in or the user can opt-out according to the local regulation that applies. And consequently, regardless of whether the use of "client credentials" is allowed.

That sentence, was recently included in https://github.com/camaraproject/IdentityAndConsentManagement/pull/121/commits/be812b8a9dbafd96733fbdbbdc85a75055d4fd47. It was later added to most of the WG participants' reviews. Perhaps you can leave it out for now.

Reply:
@jpengar: your statement is exactly in line with what I want to express. We need to phrase the sentence in a way that the purpose must be declared in the access token request (and a 3 legged access token must be used), every time personal data is processed by an API and a user can exercise his rights either via opt-in and/or opt-out (according to GDPR or other legislations).

Reply:
I tried to compile @Elisabeth-Ericsson "suggestion" and @jpengar comment into a new text.
At first I failed because I mixed up legal-basis and legislation. With @hdamker help we propose now this text.
https://github.com/AxelNennker/IdentityAndConsentManagement/blob/camara_oidc_profile/documentation/CAMARA-Security-Interoperability.md#purpose

> In Camara purpose is used to convey to the user for which purpose an API is used. E.g.: Does the client request the user's location because it wants to show relevant tourist information or is the user's location information protecting the user when they withdraw money from an ATM?

> Purpose has always to be reflected and audited when personal information is accessed or managed in certain legislations (e.g. in EU GDPR). To ensure interoperability across different legislations this profile requires that the purpose must always be declared within CIBA or OIDC Authorization Code flow token requests if personal information is accessed or managed.

Reply:
Client credentials must also be considered, unless we restrict those scenarios that access / manage personal information to not use this grant type, even if no opt-out/opt-in is allowed (related discussion about this restriction, proposed to be lifted: https://github.com/camaraproject/IdentityAndConsentManagement/issues/131)

Reply:
@diegogonmar I have intentionally proposed to not mentioned Client credentials here, for two reasons:

* Client Credential flow SHOULD NOT be used if personal information is accessed or managed ... therefore the formulation "the purpose must always be declared within CIBA or OIDC Authorization Code flow token requests if personal information is accessed or managed", mentioning Client Credentials here would give the wrong impression that this is a valid flow for requesting access to personal information.
* in the exceptional case that Client Credentials are used and personal information is accessed or managed, than this has been "agreed between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API" (cf. the Client Credential flow section here in the profile) - so the purpose is already declared upfront. Nevertheless API Client and the Telco Operator can also include in the agreement, that the purpose has to be send in the scope. BTW has the Client Credential flow no authorisation request, just the client authentication which is answered by an access token.

Beyond that I would be fine with adding "It SHOULD NOT be used if personal information is accessed or managed" to the Client Credential section.

Reply:
Unless we forbid Client Credentials, and _should not_ is not forbidding it, we need to declare in the profile that when client credentials is used and manages personal information, purpose needs to be used and the solution we finally choose applies as well. 
The second bullet in your text can apply also to AuthCode or CIBA, i.e.: the API Client and Telco Operator may agree as well the purpose to be used by an API Client, and agree to use e.g.: CIBA. In any case, no matter the flow used, purpose is needed. If we agree for AuthCode/CIBA, it applies to Client Credentials.

Managing of personal information and therefore need to indicate the purpose is not related with one flow or another, it would apply the same even if we use non OAuth/OIDC flows.

Reply:
Personally I think that Authorization Code Flow is the most privacy friendly and should be used whenever possible.
Client Credentials flow is at the lower end of the spectrum (there are worse like resource owner flow).
CIBA is OK-ish and in the middle.

Maybe a little hard "If you think you must use client credentials then you are doing something wrong privacy-wise".

Anyway, that does not help us here. +1 to @hdamker comment.

(Please write "suggestions" rather than comments.)

Reply:
This would be the suggestion:
_Purpose has always to be reflected and audited when personal information is accessed or managed in certain legislations (e.g. in EU GDPR). To ensure interoperability across different legislations this profile requires that the purpose must always be declared within the supported OIDC or OAuth2 token requests if personal information is accessed or managed._

Reply:
@AxelNennker 
_Maybe a little hard "If you think you must use client credentials then you are doing something wrong privacy-wise"._

This is speculative. 
If client credentials only allow purposes which do not require opt-in or opt-out, then it is just as important to reflect that fact, than when is one purpose requiring it. Because you need to guarantee privacy just in the same manner, by auditing that access to personal information was done with a purpose which did not require the opt-in or opt-out